src/Compilers/CSharp/Test/Symbol/Symbols/FunctionPointerTypeSymbolTests.cs

```diff
@@ -1102,5 +1104,218 @@ public void FormattingReturnTypeOptions()
+                // (15,9): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                //         delegate*<int, string> ptr = &M1;
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "delegate*").WithLocation(15, 9),
+                // (16,13): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
+                //         _ = ptr(1);
+                Diagnostic(ErrorCode.ERR_UnsafeNeeded, "ptr(1)").WithLocation(16, 13)
+            );
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var methodDecls = syntaxTree.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>().ToList();
+            var m2DeclSyntax = methodDecls.Skip(1).First();
+            var decls = m2DeclSyntax.DescendantNodes().OfType<VariableDeclaratorSyntax>().ToList();
+
+            var symbolInfo = model.GetSymbolInfo(decls[0].Initializer!.Value);
+            Assert.Equal("System.String C.M1(System.Int32 i)", symbolInfo.Symbol.ToTestDisplayString());
+
+            symbolInfo = model.GetSymbolInfo(decls[1].Initializer!.Value);
+            Assert.Null(symbolInfo.Symbol);
+            Assert.Single(symbolInfo.CandidateSymbols);
+            Assert.Equal("System.String C.M1(System.Int32 i)", symbolInfo.CandidateSymbols.Single().ToTestDisplayString());
+            Assert.Equal(CandidateReason.OverloadResolutionFailure, symbolInfo.CandidateReason);
+
+            var invocations = m2DeclSyntax.DescendantNodes().OfType<InvocationExpressionSyntax>().ToList();
+            Assert.Equal(2, invocations.Count);
+
+            symbolInfo = model.GetSymbolInfo(invocations[0]);
+            Assert.Empty(symbolInfo.CandidateSymbols);
+            Assert.Equal("delegate*<System.Int32, System.String>", symbolInfo.Symbol.ToTestDisplayString());
+
+            symbolInfo = model.GetSymbolInfo(invocations[1]);
+            Assert.Null(symbolInfo.Symbol);
+            Assert.Equal("delegate*<System.Int32, System.String>", symbolInfo.CandidateSymbols.Single().ToTestDisplayString());
+            Assert.Equal(CandidateReason.OverloadResolutionFailure, symbolInfo.CandidateReason);
+
+            var m3DeclSyntax = methodDecls.Skip(2).Single();
+            symbolInfo = model.GetSymbolInfo(m3DeclSyntax.DescendantNodes().OfType<VariableDeclaratorSyntax>().Single().Initializer!.Value);
+            Assert.Empty(symbolInfo.CandidateSymbols);
+            Assert.Equal("System.String C.M1(System.Int32 i)", symbolInfo.Symbol.ToTestDisplayString());
+
+            symbolInfo = model.GetSymbolInfo(m3DeclSyntax.DescendantNodes().OfType<InvocationExpressionSyntax>().Single());
+            Assert.Empty(symbolInfo.CandidateSymbols);
+            Assert.Equal("delegate*<System.Int32, System.String>", symbolInfo.Symbol.ToTestDisplayString());
+        }
```

Comment:
>} [](start = 8, length = 1)

For both tests, it is hard to tell whether all interesting scenarios are covered. Instead of trying to create new tests focused solely on SemanticModel, I would modify a set of existing tests (a set of tests covering the full breadth of "unique" success and error scenarios). I would also cover the same set of scenarios for IOperation.

Reply:
I've gone through and added a bunch more testing to existing scenario tests. I haven't done every single variation as I didn't think there were meaningful differences from a public API perspective, but I believe I've handled the majority of the interesting public api cases.

---
In reply to: [428933056](https://github.com/dotnet/roslyn/pull/44436#discussion_r428933056) [](ancestors = 428933056)